File: 1299.将每个元素替换为右侧最大元素.js
/*
 * @lc app=leetcode.cn id=1299 lang=javascript
 *
 * [1299] 将每个元素替换为右侧最大元素
 */

// @lc code=start
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    const n = arr.length;
    const ans = new Array(n);
    ans[n - 1] = -1;
    for (let i = n - 2; i >= 0; i--) {
        ans[i] = Math.max(ans[i + 1], arr[i + 1]);
    }
    return ans;
};
// @lc code=end

File: 1299.将每个元素替换为右侧最大元素.java
/*
 * @lc app=leetcode.cn id=1299 lang=java
 *
 * [1299] 将每个元素替换为右侧最大元素
 */

// @lc code=start
class Solution {
    public int[] replaceElements(int[] arr) {
        // 从右边向左操作,并记录右边的最大值
        // 然后用最大值替换左边的值并更新最大值

        int n = arr.length;
        int[] ans = new int[n];
        ans[n - 1] = -1;
        for (int i = n - 2; i >= 0; --i) {
            ans[i] = Math.max(ans[i + 1], arr[i + 1]);
        }
        return ans;
    }
}
// @lc code=end

